src/core/std_ranges.h

```diff
@@ -0,0 +1,26 @@
+// Copyright 2025, DragonflyDB authors.  All rights reserved.
+// See LICENSE for licensing terms.
+//
+
+#pragma once
+
+#include <algorithm>
+
+namespace rng {
+
+namespace detail {
+
+#define WRAPPER(name, algo)                                                \
+  template <typename C, typename... Args> auto name(C&& c, Args... args) { \
+    return algo(c.begin(), c.end(), std::forward<Args>(args)...);          \
+  }
+
+WRAPPER(all_of, std::all_of);
+WRAPPER(sort, std::sort);
+
+}  // namespace detail
+
```

Comment:
@dranikpg The idea is to quickly provide function wrappers for all `std::algorithms`. 

That way we can call `rng::algo(cont, predicate)` as seen in the examples. 

WDYT?